.github/action-common-python-release/action.yml

```
@@ -0,0 +1,72 @@
+name: "Base Python Release"
+description: "Common tasks required to release python wheels on any platform"
+
+inputs:
+  package-name:
+    required: true
+    description: "Either `qcs-sdk-python` or `qcs-sdk-python-grpc-web`"
+
+  python-version:
+    default: "3.12"
+    description: "Python version used for script steps"
+  python-architecture:
+    description: "Python architecture used for script steps"
+  rust-target:
+    description: "Rust target used for script steps"
+  maturin-command:
+    default: build
+    description: "Maturin command used, `build` or `sdist`"
+  maturin-args:
+    description: "Additional maturin command arguments"
+  maturin-target:
+    description: "Rust target used for maturin compilation"
+
+
+runs:
+  using: "composite"
+  steps:
+  - uses: actions/setup-python@v4
+    with:
+      python-version: ${{ inputs.python-version }}
+      architecture: ${{ inputs.python-architecture }} 
+  - name: Install Rust toolchain 
+    uses: actions-rs/toolchain@v1 
+    with: 
+      toolchain: stable 
+      target: ${{ inputs.rust-target }}
+      profile: minimal 
+      default: true 
+  - name: Patch package metadata for grpc-web
+    shell: bash
+    if: matrix.package-name == 'qcs-sdk-python-grpc-web'
+    run: |
+      pip install toml
+      python crates/python/scripts/patch_grpc_web.py
+      cargo update
```

`cargo update` might update more than we need for this step, e.g. anything with "SemVer-compatible" updates. We can restrict it to specific packages using the `-p` flag.

```suggestion
      cargo update -p hyper-proxy -p qcs-sdk-python-grpc-web
```

Note: this will probably require also patching the workspace `Cargo.toml`

---

Couldn't use `-p` and `SPEC` at the same time, so changed to `cargo update hyper-proxy`